src/Symfony/Component/Console/Application.php

```diff
@@ -466,7 +466,9 @@ public function get($name)
             $command = $this->commands[$name];
         } elseif ($this->commandLoader && $this->commandLoader->has($name)) {
             $command = $this->commandLoader->get($name);
-            $command->setName($name);
+            if (!$command->getName()) {
+                $command->setName($name);
+            }
             $this->add($command);
```

Comment:
I agree it would be appropriated, but this logic is in `add()` for years, any custom implementation doing `parent::add($command);$command->getApplication()->foo()` would break, I think it's not worth changing this now